crates/matrix-sdk-crypto/src/error.rs

```diff
@@ -70,6 +70,35 @@ pub enum OlmError {
             have a valid Olm session with us"
     )]
     MissingSession,
+
+    #[error(transparent)]
+    /// The room key that was to be shared was not shared because the sharing
+    /// strategy could not be fulfilled.
+    RoomKeySharingStrategyError(RoomKeySharingStrategyError),
```

Comment:
#3810 added a "SessionRecipientCollectionError" which I think we should be using here, rather than adding a new error type.